omegaconf/dictconfig.py

```diff
@@ -662,6 +672,9 @@ def _dict_conf_eq(d1: "DictConfig", d2: "DictConfig") -> bool:
         assert isinstance(d2, DictConfig)
         if len(d1) != len(d2):
             return False
+        if d1._is_missing() or d2._is_missing():
+            return d1._is_missing() is d2._is_missing()
```

Comment:
So, to clarify, my suggestion to use `and` wasn't related to the speed of `==`: it was a cheap way to avoid computing `d2._is_missing()` if `d1._is_missing()` was False.
(ideally both would be called only once, but until we can use the new `:=` operator it's probably not worth it as it would make the code more complex for very little gain)